### ISSUE TITLE
QR scan does not work

### ISSUE THREAD

**Issue by hellpe** [X1nto/Mauth#43]
When aiming at a QR code, the camera view is displayed but the scan does not occur.

Tested with PayPal and Firefox Accounts on Android 12 (Crosscall Action-X5), with Mauth 0.5.3 (F-Droid)

**Comment by X1nto** [comment]
Could you try rotating your phone? I heard there's an issue with some QR codes requiring rotation

**Comment by hellpe** [comment]
> Could you try rotating your phone? I heard there's an issue with some QR codes requiring rotation

Rotating the phone does not fix the issue, either with or without auto-rotate

**Comment by LyesSaadi** [comment]
Same issue here ! Trying to scan a QRCode from my computer's Authenticator app (https://flathub.org/apps/com.belmoussaoui.Authenticator), and it does not work. Other QR Code scanning apps can scan it successfully.

**Comment by X1nto** [comment]
Tested on my device and Firefox works just fine with the latest version. Unfortunately, there's not much I can do.

**Comment by LyesSaadi** [comment]
Decided to try something quite absurd as a last resort by printing the QR Code and trying to read it, thinking it maybe was an issue with the computer screen. And, nope, not working :/...

**Comment by LyesSaadi** [comment]
Also, ZXing (by installing org.google.zxing.client.android.apk from F-Droid) does work in reading the QR Code without issue, so it doesn't seem to be a library issue.

**Comment by X1nto** [comment]
> Decided to try something quite absurd as a last resort by printing the QR Code and trying to read it, thinking it maybe was an issue with the computer screen. And, nope, not working :/...

I think the issue might've been with the camera library. Could you try with v0.6.0 I released yesterday? It should include the updated CameraX library.

**Comment by LyesSaadi** [comment]
That does not solve it unfortunately :/...

**Comment by X1nto** [comment]
I'm afraid I can't exactly fix the issue then. Everything works just fine on my devices. If you have any specific QR Codes which work on other authenticators but don't work on Mauth, feel free to let me know. (I'm asking for specific QR Codes because some have a slightly different format of the OTP string which Mauth may not support, so this could help me add support for more and more authentication types)

**Comment by LyesSaadi** [comment]
Okay, so, the URI I get from scanning the QR Code is in this format:
```
otpauth://totp/account?secret=secret&issuer=issuer&algorithm=algorithm&digits=digits&period=period
```

**Comment by X1nto** [comment]
Could you make it so the only "censored" parts are the secret and the issuer? everything else is very essential for me to test how the parsing works

**Comment by LyesSaadi** [comment]
```
otpauth://totp/account?secret=secret&issuer=issuer&algorithm=sha1&digits=6&period=30
```
(I also censored the account since it would reveal the issuer in my case)

**Comment by X1nto** [comment]
> ```
> otpauth://totp/account?secret=secret&issuer=issuer&algorithm=sha1&digits=6&period=30
> ```
> 
> (I also censored the account since it would reveal the issuer in my case)

Oh, I think I found the culprit. The code checks for "SHA1", not "sha1". I'll push the fix to v6.0.1.

**Comment by X1nto** [comment]
> I can confirm that's the issue by creating a new QR Code with the SHA1 all caps !

Tested it myself as well (and added instrumentation tests for automatization, should've done it a looooooooong time ago), and works just fine. Can you confirm that scanning works just fine now?
